projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -194,11 +198,51 @@ export class ExplorerComponent {
     }
   }
 
+  private convertToFilterAttributes(attrArray: AttributeMetadata[]): FilterAttribute[] {
+    return attrArray.map(({ name, displayName, units, type, onlySupportsAggregation, onlySupportsGrouping }) => {
+      const applicableType = toFilterAttributeType(type);
+
+      return {
+        name: name,
+        displayName: displayName,
+        units: units,
+        type: applicableType,
+        onlySupportsAggregation: onlySupportsAggregation,
+        onlySupportsGrouping: onlySupportsGrouping
+      };
+    });
+  }
+
   public onContextUpdated(contextWrapper: ExplorerContextScope): void {
     this.attributes$ = this.metadataService.getFilterAttributes(contextWrapper.dashboardContext);
+    const listener = this.attributes$.subscribe(attributes => {
```

Comment:
We shouldn't use a subscription here. Make the filters an observable, and subscribe via async pipe in the template.